### PR TITLE
1418-cleanup-warnings-0020

### PR DESCRIPTION
**Pull request by giduac**
1418-cleanup-warnings-0020
[issue 1418](https://github.com/Krypton-Suite/Standard-Toolkit/issues/1418)

This concludes the Workspace module.

compile-results

### CODE REVIEW

File: Source/Krypton Components/Krypton.Workspace/Controls Workspace/KryptonWorkspace.cs [hunk 1]
```diff
@@ -2804,7 +2804,7 @@ internal Rectangle SeparatorMoveBox(ViewDrawWorkspaceSeparator separator)
                                      Point splitter)
         {
             // Get the sequence that contains the items moved
-            var parentSequence = (KryptonWorkspaceSequence)separator.WorkspaceItem.WorkspaceParent;
+            var parentSequence = separator.WorkspaceItem.WorkspaceParent as KryptonWorkspaceSequence;
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Workspace/Controls Workspace/KryptonWorkspace.cs:2807]: If the var is just going to be ignoring the nullable possibility from using the `as`
then use a cast, and remove the `!` usage later on.

File: Source/Krypton Components/Krypton.Workspace/Controls Workspace/KryptonWorkspace.cs [hunk 2]
```diff
@@ -2899,7 +2899,7 @@ internal void InternalPageDragStart(object sender, KryptonNavigator? navigator,
                     }
 
                     // Calling DragStart will cause the drag targets to be created from the target providers
-                    e.Cancel = !_dragManager.DragStart(e.ScreenPoint, new PageDragEndData(sender as KryptonNavigator, e.Pages));
+                    e.Cancel = !_dragManager.DragStart(e.ScreenPoint, new PageDragEndData((sender as KryptonNavigator)!, e.Pages));
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Workspace/Controls Workspace/KryptonWorkspace.cs:2902]: Either 
- check the nullability from the `as` before sending
- DragStart to accept null's
- Use a cast.

Option 2 preferred.

**Review by Smurf-IV — CHANGES_REQUESTED** [review]
Apply comments on casting / null check / null ignore `!` override to all code changes please ?

File: Source/Krypton Components/Krypton.Workspace/Controls Workspace/KryptonWorkspace.cs (hunk 2 above)
Comment (giduac) [reply Source/Krypton Components/Krypton.Workspace/Controls Workspace/KryptonWorkspace.cs:2902]: @Smurf-IV 
Dragstart already does a null check on the dragEndData param and will throw if null.
The PageDragEndData class expects a valid source param.

This way the sender is checked and dragstart handles the throw
```cs
PageDragEndData? dragEndData = sender is null
    ? null
    : new PageDragEndData((sender as KryptonNavigator)!, e.Pages);

e.Cancel = !_dragManager.DragStart(e.ScreenPoint, dragEndData);
```

Your thoughts please ?

Comment (Smurf-IV) [reply Source/Krypton Components/Krypton.Workspace/Controls Workspace/KryptonWorkspace.cs:2902]: The `sender is null` should have the following check in it `(sender as KryptonNavigator)`
ReSharper will then change it to a null check pattern along the lines of `(sender is KryptonNavigator kn)` and pass the `kn` into the constructor.

**Pull request merged**